# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/windows/nsDragService.cpp
# Commit: beeda5ef3712
# Full Hash: beeda5ef3712b73a19c7a4ce08a52ab4dd08b802
# Author: David P <daparks@mozilla.com>
# Date: 2024-07-04 16:09:52
# Regressor Bug: 1893119
# File Overlap Count: 1
# Description:
#   Bug 1893119: Part 20 - Move EndDragSession from nsIDragService to nsIDragSession r=gstoll,rkraesig,win-reviewers,geckoview-reviewers,m_kato
#   
#   This is a straightforward move.  It does take the liberty of breaking out an
#   EndDragSessionImpl method, which will be needed later.
#   
# ==============================================================================

diff -r 6c3ffc8c534c -r beeda5ef3712 widget/windows/nsDragService.cpp
--- a/widget/windows/nsDragService.cpp	Thu Jul 04 07:48:10 2024 +0000
+++ b/widget/windows/nsDragService.cpp	Thu Jul 04 07:48:11 2024 +0000
@@ -310,8 +310,11 @@
   }
   SetDragEndPoint(LayoutDeviceIntPoint(cpos.x, cpos.y));
 
-  ModifierKeyState modifierKeyState;
-  EndDragSession(true, modifierKeyState.GetModifiers());
+  RefPtr<nsIDragSession> session = GetCurrentSession(aWidget);
+  if (session) {
+    ModifierKeyState modifierKeyState;
+    session->EndDragSession(true, modifierKeyState.GetModifiers());
+  }
 
   mDoingDrag = false;
 
@@ -615,8 +618,8 @@
 // when the drag ends. It seems that OLE doesn't like to let apps quit
 // w/out crashing when we're still holding onto their data
 //
-NS_IMETHODIMP
-nsDragService::EndDragSession(bool aDoneDrag, uint32_t aKeyModifiers) {
+nsresult nsDragSession::EndDragSessionImpl(bool aDoneDrag,
+                                           uint32_t aKeyModifiers) {
   // Bug 100180: If we've got mouse events captured, make sure we release it -
   // that way, if we happen to call EndDragSession before diving into a nested
   // event loop, we can still respond to mouse events.
@@ -624,7 +627,7 @@
     ::ReleaseCapture();
   }
 
-  nsBaseDragService::EndDragSession(aDoneDrag, aKeyModifiers);
+  nsBaseDragSession::EndDragSessionImpl(aDoneDrag, aKeyModifiers);
   NS_IF_RELEASE(mDataObject);
 
   return NS_OK;